pptx_blueprint/__init__.py

```diff
@@ -85,8 +90,38 @@ def _find_shapes_in_slide(slide):
     def save(self, filename: _Pathlike) -> None:
         """Saves the updated pptx to the specified filepath.
 
-        Args:
+         Args:
             filename (path-like): file name or path
         """
         # TODO: make sure that the user does not override the self._template_path
         pass
+
+    def save_pdf(self, file_path: _Pathlike) -> None:
+        """Exports the updated pptx to the specified filepath as pdf file.
+
+        Args:
+            file_path (path-like) file name or path
+        """
+
+        try:
+            subprocess.run(['libreoffice', '--version'],
+                           stdout=subprocess.DEVNULL)  # check if libreoffice is installed
+
+            path = Path(file_path)
+            outdir = path.parent
+            file_name = path.name
+
+            # create temporary  directory
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                template_temporary_path = f'{tmpdirname}/{file_name}'
+
+                # save current Template as pptx in temporary directory
+                # TODO replace with self.save() method
+                self._presentation.save(template_temporary_path)
+
+                export_cmd = ['libreoffice', '--headless', '--convert-to',
+                              'pdf', '--outdir', outdir, template_temporary_path]
+                p = subprocess.run(
```

Comment:
It would be cool if we could do some error handling. Unfortunately, on my machine, the exit code is 0 regardless of the file being converted or not. Still, one way would be to capture the `stderr` and check for error messages:

stderr of a successful run:
```
javaldx: Could not find a Java Runtime Environment!
Warning: failed to read path from javaldx
```

stderr of an erroneous run:
```
javaldx: Could not find a Java Runtime Environment!
Warning: failed to read path from javaldx
Error: source file could not be loaded
```

Reply:
Since we don't know what to look for in `stderr` and it also contains output when being successful, investigating `stderr` is difficult.

Instead, I propose to check if the intended output file exists. If do something like
~~~
raise ConversionError(
    f'Conversion to PDF using LibreOffice failed with error: {stderr}')
~~~

Reply:
Yes, checking whether the PDF file exists is a far simpler (and hence superior) idea. :+1: